Docker Build Fails

When I try to build the docker container (see log below), the build fails.

Basically I only clone the repository and docker-build it. 

(I was going back to a commit SHA that worked for me earlier) - I successfully built the image on the 23rd of Sept at 9:57 GMT+2 using the same CI setup.

```
$ git clone https://github.com/mzbench/mzbench
Cloning into 'mzbench' ...
$ cd mzbench
$ git checkout f84b78e8068cb26c7e5cbbcfec41eb9a1f0e46f4

HEAD is now at f84b78e... Updating logging
$ docker build -t $REPOSITORY_BASE/mzbench .
Sending build context to Docker daemon  30.64MB

Step 1/25 : FROM erlang:20.3.8-alpine as build
 ---> ac4c7b98211c
Step 2/25 : ENV ORIG_PATH=$PATH
 ---> Using cache
 ---> b8916a6e0f5b
Step 3/25 : ENV CODE_LOADING_MODE=interactive
 ---> Using cache
 ---> acedc24f4298
Step 4/25 : ENV MZBENCH_SRC_DIR /opt/mzbench_src
 ---> Using cache
 ---> 2baabf9be28a
Step 5/25 : ENV MZBENCH_API_DIR /opt/mzbench_api
 ---> Using cache
 ---> 1974a8fa54d4
Step 6/25 : ENV HOME_DIR /root
 ---> Using cache
 ---> 4ff79b6fc99a
Step 7/25 : WORKDIR $MZBENCH_SRC_DIR
 ---> Using cache
 ---> 7eb0490bb98c
Step 8/25 : RUN apk add --no-cache g++ make musl-dev zlib-dev py2-pip openssl git rsync
 ---> Using cache
 ---> 4e2752c0c895
Step 9/25 : COPY . .
 ---> 73502f7038e2
Step 10/25 : RUN mkdir -p ${HOME_DIR}/.local/share/mzbench_workers     && pip install -r requirements.txt     && make -C ./server generate     && cp -R ./server/_build/default/rel/mzbench_api ${MZBENCH_API_DIR}/     && make -C ./node install     && make -C ./node local_tgz     && ln -s ${HOME_DIR}/.local/cache/mzbench_api/packages/node-*_erts*.tgz ${HOME_DIR}/.local/cache/mzbench_api/packages/node-someversion-someos.tgz     && ln -s ${HOME_DIR}/.local/cache/mzbench_api/packages/node-*_erts*.tgz ${HOME_DIR}/.local/cache/mzbench_api/packages/node-$(git rev-parse HEAD)-someos.tgz
 ---> Running in fb85f2db6b16
Collecting docopt>=0.6.2 (from -r requirements.txt (line 2))
  Retrying (Retry(total=4, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.VerifiedHTTPSConnection object at 0x7fc5d1450690>: Failed to establish a new connection: [Errno -3] Try again',)': /packages/a2/55/8f8cab2afd404cf578136ef2cc5dfb50baa1761b68c9da1fb1e4eed343c9/docopt-0.6.2.tar.gz
  Downloading https://files.pythonhosted.org/packages/a2/55/8f8cab2afd404cf578136ef2cc5dfb50baa1761b68c9da1fb1e4eed343c9/docopt-0.6.2.tar.gz
Collecting requests>=2.7.0 (from -r requirements.txt (line 3))
  Downloading https://files.pythonhosted.org/packages/51/bd/23c926cd341ea6b7dd0b2a00aba99ae0f828be89d72b2190f27c11d4b7fb/requests-2.22.0-py2.py3-none-any.whl (57kB)
Collecting erl_terms>=0.1.1 (from -r requirements.txt (line 4))
  Downloading https://files.pythonhosted.org/packages/78/74/f1fae072c9bce0ea013698d176254c0b20f129a54235b29f0ca050697aad/erl_terms-0.1.5.tar.gz
Collecting chardet<3.1.0,>=3.0.2 (from requests>=2.7.0->-r requirements.txt (line 3))
  Downloading https://files.pythonhosted.org/packages/bc/a9/01ffebfb562e4274b6487b4bb1ddec7ca55ec7510b22e4c51f14098443b8/chardet-3.0.4-py2.py3-none-any.whl (133kB)
Collecting idna<2.9,>=2.5 (from requests>=2.7.0->-r requirements.txt (line 3))
  Downloading https://files.pythonhosted.org/packages/14/2c/cd551d81dbe15200be1cf41cd03869a46fe7226e7450af7a6545bfc474c9/idna-2.8-py2.py3-none-any.whl (58kB)
Collecting urllib3!=1.25.0,!=1.25.1,<1.26,>=1.21.1 (from requests>=2.7.0->-r requirements.txt (line 3))
  Downloading https://files.pythonhosted.org/packages/e0/da/55f51ea951e1b7c63a579c09dd7db825bb730ec1fe9c0180fc77bfb31448/urllib3-1.25.6-py2.py3-none-any.whl (125kB)
Collecting certifi>=2017.4.17 (from requests>=2.7.0->-r requirements.txt (line 3))
  Downloading https://files.pythonhosted.org/packages/18/b0/8146a4f8dd402f60744fa380bc73ca47303cccf8b9190fd16a827281eac2/certifi-2019.9.11-py2.py3-none-any.whl (154kB)
Collecting parsimonious==0.7.0 (from erl_terms>=0.1.1->-r requirements.txt (line 4))
  Downloading https://files.pythonhosted.org/packages/11/db/06a1d0a41b4d236cd84fb27fced4479645b1fb6100501e03accde2804e51/parsimonious-0.7.0.tar.gz
Collecting six (from parsimonious==0.7.0->erl_terms>=0.1.1->-r requirements.txt (line 4))
  Downloading https://files.pythonhosted.org/packages/73/fb/00a976f728d0d1fecfe898238ce23f502a721c0ac0ecfedb80e0d88c64e9/six-1.12.0-py2.py3-none-any.whl
Installing collected packages: docopt, chardet, idna, urllib3, certifi, requests, six, parsimonious, erl-terms
  Running setup.py install for docopt: started
    Running setup.py install for docopt: finished with status 'done'
  Running setup.py install for parsimonious: started
    Running setup.py install for parsimonious: finished with status 'done'
  Running setup.py install for erl-terms: started
    Running setup.py install for erl-terms: finished with status 'done'
Successfully installed certifi-2019.9.11 chardet-3.0.4 docopt-0.6.2 erl-terms-0.1.5 idna-2.8 parsimonious-0.7.0 requests-2.22.0 six-1.12.0 urllib3-1.25.6
make: Entering directory '/opt/mzbench_src/server'
rm -rf .make/git-rev-*
mkdir -p .make && touch .make/git-rev-f84b78e8068cb26c7e5cbbcfec41eb9a1f0e46f4
/opt/mzbench_src/bin/rebar3 compile
===> Fetching rebar_rsync_deps (from {git,"https://github.com/ethercrow/rebar3_rsync_resource",
                            {ref,"53eb6aa50db8e0de7cf5833a48e48c205f5c8d56"}})
===> Compiling rebar_rsync_deps
_build/default/plugins/rebar_rsync_deps/src/rebar_rsync_resource.erl:14: Warning: variable 'Dir' is unused
_build/default/plugins/rebar_rsync_deps/src/rebar_rsync_resource.erl:17: Warning: variable 'State' is unused
_build/default/plugins/rebar_rsync_deps/src/rebar_rsync_resource.erl:29: Warning: variable 'Dir' is unused

===> Using custom resource rebar_rsync_resource that implements a deprecated api. It should be upgraded to rebar_resource_v2.
===> Using custom resource rebar_rsync_resource that implements a deprecated api. It should be upgraded to rebar_resource_v2.
===> Verifying dependencies...
===> Fetching cowboy (from {git,"git://github.com/ninenines/cowboy.git",
                  {ref,"90ae31998e8d0887b9efe4b441136ac047708bb9"}})
===> Fetching erlcloud (from {git,"git://github.com/erlcloud/erlcloud.git",{tag,"3.1.7"}})
===> Fetching gen_smtp (from {git,"git://github.com/Vagabond/gen_smtp.git",{tag,"0.12.0"}})
===> Fetching hamcrest (from {git,"git://github.com/hyperthunk/hamcrest-erlang",
                    {ref,"a8578932014c86fe395982fa3151d48897b9c37f"}})
===> Failed to update package header_generator from repo hexpm
===> Plugin header_generator not available. It will not be used.
===> Failed to update package eqc_resolver from repo hexpm
===> Plugin eqc_resolver not available. It will not be used.
===> Fetching hdr_histogram (from {git,"git://github.com/machinezone/hdr_histogram_erl.git",
                         {tag,'experimental-multithread-support2'}})
===> Fetching jiffy (from {git,"git://github.com/davisp/jiffy.git",{tag,"0.14.2"}})
===> Failed to update package rebar_gdb_plugin from repo hexpm
===> Plugin rebar_gdb_plugin not available. It will not be used.
===> Fetching lager (from {git,"git://github.com/erlang-lager/lager.git",{tag,"3.8.0"}})
===> Fetching meck (from {git,"git://github.com/eproxus/meck.git",{tag,"0.8.13"}})
===> Fetching mzbench_language (from {rsync,"../common_apps/mzbench_language/"})
===> Fetching mzbench_utils (from {rsync,"../common_apps/mzbench_utils/"})
===> Fetching base16 v1.0.0
===> Downloaded package, caching at /root/.cache/rebar3/hex/hexpm/packages/base16-1.0.0.tar
===> Fetching cowlib (from {git,"git://github.com/ninenines/cowlib.git","1.2.0"})
===> WARNING: It is recommended to use {branch, Name}, {tag, Tag} or {ref, Ref}, otherwise updating the dep may not work as expected.
===> Fetching edown (from {git,"https://github.com/uwiger/edown.git",
                 {ref,"b7c8eb0ac1859f8fce11cbfe3526f5ec83194776"}})
===> Fetching eini v1.2.4
===> Downloaded package, caching at /root/.cache/rebar3/hex/hexpm/packages/eini-1.2.4.tar
===> Fetching fqc (from {git,"https://github.com/project-fifo/fqc.git",
               {ref,"ad62f8121414ed026a6a9f564b1856191ab7d1fc"}})
===> Fetching goldrush v0.1.9
===> Downloaded package, caching at /root/.cache/rebar3/hex/hexpm/packages/goldrush-0.1.9.tar
===> Fetching jsx v2.8.0
===> Downloaded package, caching at /root/.cache/rebar3/hex/hexpm/packages/jsx-2.8.0.tar
===> Fetching lhttpc v1.5.4
===> Downloaded package, caching at /root/.cache/rebar3/hex/hexpm/packages/lhttpc-1.5.4.tar
===> Skipping mzbench_utils (from {rsync,"../mzbench_utils/"}) as an app of the same name has already been fetched
===> Fetching pretty_errors (from {git,"git://github.com/ethercrow/pretty_errors",
                         "531d79b0057c3b08b8509b326b9bdaf663bf45ce"})
===> WARNING: It is recommended to use {branch, Name}, {tag, Tag} or {ref, Ref}, otherwise updating the dep may not work as expected.
===> Fetching ranch (from {git,"git://github.com/ninenines/ranch.git","1.0.0"})
===> WARNING: It is recommended to use {branch, Name}, {tag, Tag} or {ref, Ref}, otherwise updating the dep may not work as expected.
===> Compiling cowlib
_build/default/deps/cowlib/src/cow_multipart.erl:392: Warning: call to crypto:rand_bytes/1 will fail, since it was removed in 20.0; use crypto:strong_rand_bytes/1

===> Compiling ranch
===> Compiling lhttpc
_build/default/deps/lhttpc/src/lhttpc_manager.erl:270: Warning: call to crypto:rand_bytes/1 will fail, since it was removed in 20.0; use crypto:strong_rand_bytes/1

===> Compiling jsx
===> Compiling eini
===> Compiling base16
===> Compiling erlcloud
===> Compiling fqc
===> "/opt/mzbench_src/server/_build/default/deps/fqc/ebin/fqc.app" is missing kernel from applications list
===> "/opt/mzbench_src/server/_build/default/deps/fqc/ebin/fqc.app" is missing stdlib from applications list
===> Compiling gen_smtp
_build/default/deps/gen_smtp/src/smtp_util.erl:71: Warning: crypto:rand_uniform/2 is deprecated and will be removed in a future release; use rand:uniform/1
_build/default/deps/gen_smtp/src/smtp_util.erl:71: Warning: crypto:rand_uniform/2 is deprecated and will be removed in a future release; use rand:uniform/1

===> Compiling pretty_errors
===> Compiling mzbench_utils
===> Compiling _build/default/deps/mzbench_utils/src/mzb_lists.erl failed
_build/default/deps/mzbench_utils/src/mzb_lists.erl:37: illegal pattern
_build/default/deps/mzbench_utils/src/mzb_lists.erl:37: variable 'E' is unbound
_build/default/deps/mzbench_utils/src/mzb_lists.erl:37: variable 'ST' is unbound

make: *** [Makefile:87: .make/compilation-up-to-date] Error 1
make: Leaving directory '/opt/mzbench_src/server'
The command '/bin/sh -c mkdir -p ${HOME_DIR}/.local/share/mzbench_workers     && pip install -r requirements.txt     && make -C ./server generate     && cp -R ./server/_build/default/rel/mzbench_api ${MZBENCH_API_DIR}/     && make -C ./node install     && make -C ./node local_tgz     && ln -s ${HOME_DIR}/.local/cache/mzbench_api/packages/node-*_erts*.tgz ${HOME_DIR}/.local/cache/mzbench_api/packages/node-someversion-someos.tgz     && ln -s ${HOME_DIR}/.local/cache/mzbench_api/packages/node-*_erts*.tgz ${HOME_DIR}/.local/cache/mzbench_api/packages/node-$(git rev-parse HEAD)-someos.tgz' returned a non-zero code: 2
ERROR: Job failed: exit status 1
```

--- a/Dockerfile
+++ b/Dockerfile
@@ -1,5 +1,5 @@
 ### --- Build mzbench artifacts --- ###
-FROM erlang:20.3.8-alpine as build
+FROM erlang:22.1.8-alpine as build
 
 ENV ORIG_PATH=$PATH
 ENV CODE_LOADING_MODE=interactive

